src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,97 @@
+package duke.task;
+
+import duke.exception.DukeException;
+
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The type Deadline which extends from Task.
+ * Handles all inputs which has "deadline..."
+ */
+public class Deadline extends Task {
+
+    // by is completion date.
+    // d1 is the date that is being parsed
+    // Time is the deadline completion time (IF THERE IS!)
+    private String by;
+
+
+    /**
+     * Instantiates a new Deadline.
+     *
+     * @param description the description
+     */
+    public Deadline(String description) {
+        super(description);
+    }
+
+    /**
+     * Sets the date and time of the deadline task.
+     *
+     * @param by the by
+     */
+    public void setBy(String by) {
+        this.by = super.set_by_at(by);
+    }
+
+    private String getBy() {
+        return by;
+    }
+
+    /**
+     * Sets d1, the LocalDateTime for the Task.
+     *
+     * @throws DukeException the duke exception
+     */
+    public void setD1() throws DukeException {
+        super.setD1(getBy());
+    }
+
+    /**
+     * From the input given by the user, filter out the commands (Deadline)
+     * And returns the description of the string.
+     *
+     * @param s the s
+     * @return String without "deadline"
+     * @throws DukeException when the user enters an empty description or use the wrong deadline format
+     */
+
+    //Get the task from the given input.
+    @Override
+    public String formatTasks(String s) throws DukeException {
+        String[] splitedString = s.split("deadline ");
+        if (splitedString.length < 1) {
+            throw new DukeException("You cannot leave the description empty");
+        } else {
```

Comment:
Maybe you don't need to put this part in the `else` clause, since the `if` already acts as a guard clause (you  won't be able to reach `else` if you enters `if`.